test/test_matlab_hooks.py

```diff
@@ -0,0 +1,78 @@
+import unittest
+from unittest.mock import patch
+from typing import NamedTuple
+
+import numpy as np
+
+import matlab_hooks as mh
+
+try:
+    with mh._matlab_paths():
+        import MatlabAlgorithms.MsiAlgorithms
+        import matlab
+    HAS_MCR = True
+except ImportError as ex:
+    HAS_MCR = False
+    print(ex)
+
+
+mcr_only = unittest.skipUnless(HAS_MCR, "requires MCR")
+
+
+@mcr_only
+class TestEngine(unittest.TestCase):
+    def test_initializes_engine(self):
+        mh.engine()
+
+    def test_engine_supports_divik(self):
+        engine = mh.engine()
+        self.assertTrue(hasattr(engine, 'divik'))
+
+
+@mcr_only
+class TestIsMatlab(unittest.TestCase):
+    def test_passes_matlab_data(self):
+        self.assertTrue(mh._is_matlab(matlab.double([1])))
+
+    def test_discards_numpy_data(self):
+        self.assertFalse(mh._is_matlab(np.array([1])))
+
+
+class TestDivikDefaults(unittest.TestCase):
+    def test_returns_proper_type(self):
+        defaults = mh.divik_defaults()
+        self.assertIsInstance(defaults, mh.DivikOptions)
+
+
+class TestUnroll(unittest.TestCase):
+    def test_flattens_named_tuple_into_name_value_format(self):
+        SomeTuple = NamedTuple('SomeTuple', [
+            ('blah', int),
+            ('wololo', str),
+        ])
+        a_tuple = SomeTuple(blah=1, wololo='2')
+        flat = mh._unroll(a_tuple)
+        self.assertEqual(flat.index('blah') + 1, flat.index(1))
+        self.assertEqual(flat.index('wololo') + 1, flat.index('2'))
```

Comment:
Wouldn’t `self.assertEqual(['blah', 1, 'wololo', '2'])` be clearer here? Or is there no guarantee of order? As predictable dict ordering is a recent change to CPython.

Reply:
There is no guarantee of order, unfortunately. I've been struggling this invalid preassumption on `dict`s order maaaany hours.